generator/src/main/java/com/algorand/sdkutils/listeners/TemplateGenerator.java

```diff
@@ -0,0 +1,256 @@
+            context.put("types", types);
+
+            StringWriter writer = new StringWriter();
+            template.merge( context, writer );
+
+            // TODO: Write to files instead of stdout.
+            System.out.println("====================");
+            System.out.println(StringUtils.capitalize(model.getKey().name) + ".java");
+            System.out.println("====================");
+            System.out.println(writer.toString());
+        }
+    }
+
+    private void writeQueryClass(Template t) {
+        for (QueryDef query : queries) {
+            VelocityContext context = getContext();
+
+            // Make it easier to loop thorugh parameters without caring what they are.
+            List<TypeDef> parameters = new ArrayList<>();
+            parameters.addAll(query.queryParameters);
+            parameters.addAll(query.pathParameters);
+            parameters.addAll(query.bodyParameters);
+
+            // Make it easier to get the types
+            Set<String> types = new HashSet<>();
+            for (TypeDef typeDef: parameters) {
+                types.add(typeDef.rawTypeName);
+                if (typeDef.isOfType("array")) {
+                    types.add("array");
+                }
+                if (typeDef.isOfType("enum")) {
+                    types.add("enum");
+                }
+            }
+            if (query.returnType != "String") {
+                types.add(query.returnType);
+            }
+
+            List<String> pathParts = Stream.of(query.path.split("/"))
+                            .filter(part -> !part.equals(""))
+                            .collect(Collectors.toList());
+
+            context.put("params", parameters);
+            context.put("path", pathParts);
+            context.put("types", types);
+            // TODO: Arg to inject client package.
+            context.put("package", "com.algorand.algosdk.v2.client.algod");
+
+            context.put("q", query);
+
+            StringWriter writer = new StringWriter();
+            t.merge( context, writer );
+
+            // TODO: Write to files instead of stdout.
+            System.out.println("====================");
+            System.out.println(StringUtils.capitalize(query.name) + ".java");
+            System.out.println("====================");
+            System.out.println(writer.toString());
+        }
+    }
+
+    @Override
+    public void terminate() {
+        logger.info("Generating files.");
+
+        VelocityEngine velocityEngine = new VelocityEngine();
+        Properties props = new Properties();
+        props.put("file.resource.loader.path", args.templatesDirectory.getAbsolutePath());
+        velocityEngine.init(props);
+
+        //Template modelTemplate = velocityEngine.getTemplate("model.vm");
+        //writeModelClass(modelTemplate);
+
+        Template queryTemplate = velocityEngine.getTemplate("query.vm");
+        writeQueryClass(queryTemplate);
+    }
+
```

Comment:
Below are the callbacks which trigger when the OpenAPIParser runs, they build up a series of objects that get passed into the templates.